--- a/drivers/net/ethernet/realtek/r8169.c	2019-02-27 17:08:09.000000000 +0800
+++ b/drivers/net/ethernet/realtek/r8169.c	2019-03-12 10:29:34.602334109 +0800
@@ -8414,6 +8414,11 @@
 	}
 	for (i = 0; i < ETH_ALEN; i++)
 		dev->dev_addr[i] = RTL_R8(MAC0 + i);
+	if (!is_valid_ether_addr(dev->dev_addr)) {
+		memcpy(dev->dev_addr, "\x00\xE0\x4C\x11\x22\x33", ETH_ALEN);
+		rtl_rar_set(tp, dev->dev_addr);
+		netif_info(tp, probe, dev, "r8169: set mac addr to %pM\n", dev->dev_addr);
+	}
 
 	dev->ethtool_ops = &rtl8169_ethtool_ops;
 	dev->watchdog_timeo = RTL8169_TX_TIMEOUT;
